### PR TITLE
Added test for nested directory negation pattern

### PR DESCRIPTION
**Pull request by Ferdaszewski**
Test for issue discussed in #3

This is my first pull request, so please let me know if there are any issues. Thank you!

### CODE REVIEW

**Comment by sindresorhus** [comment]
@UltCombo Any thoughts about this?

**Comment by UltCombo** [comment]
@sindresorhus #3's OP is a [node-glob issue](https://github.com/isaacs/node-glob/pull/166) regarding negation and dotfiles.

Along #3's thread another issue is discussed: it looks like `/**/*` is being treated the same as `/**`, which is most likely another issue with node-glob.

**Comment by UltCombo** [comment]
Oh wait, there's no globstar involved in the #3's OP, the actual issue is [explained by citricsquid](https://github.com/sindresorhus/del/issues/3#issuecomment-52132669) (emphasis added):

> In node-glob ** matches all children **and the parent**

So instead of:

```js
del(['public/assets/**', '!public/assets/.gitkeep'], cb);
```

You need:

```js
del(['public/assets/**', '!public/assets', '!public/assets/.gitkeep'], cb);
```

**Comment by UltCombo** [comment]
`dir/**` matches everything inside `dir` and also `dir` itself.
`dir/**/*` matches everything inside `dir`, but does not match `dir` itself.
(let's ignore the [dotfiles bug](https://github.com/isaacs/node-glob/pull/166) for now)

@isaacs Is this the expected behavior?

**Comment by UltCombo** [comment]
@sindresorhus as for the main issue, we could change globby to also negate all parent directories of each negated pattern.

However, this isn't as simple as mapping each negated pattern to multiple negated patterns when we consider complex patterns, e.g. `dir/**/foo`. I believe this would require actually matching the pattern against the filesystem, and then adding all ancestor directories of each match to the negated set of patterns.

This may be easier to implement as a node-glob option, seeing as node-glob currently handles all the matching and negated patterns logic. @isaacs Does this seem worth adding to node-glob?

File: test.js [hunk 1]
```diff
@@ -108,3 +108,37 @@ it('options are optional', function (cb) {
...
+			'tmpDir/',
+			'tmpDir/anotherDir/7.tmp',
+			'tmpDir/anotherDir/8.tmp',
+			'tmpDir/anotherDir/'
+		].forEach(fs.removeSync);
+
+	});
+
+	it('should not delete negated file', function(cb) {
+		del(['tmpDir/**/*', '!tmpDir/anotherDir/8.tmp'], function() {
```
Comment (UltCombo) [inline test.js]: `tmpDir/**/*` matches `tmpDir/anotherDir` and thus deletes the entire directory.

**Comment by sindresorhus** [comment]
> This may be easier to implement as a node-glob option, seeing as node-glob currently handles all the matching and negated patterns logic.

@UltCombo Can you open an issue on `node-glob`? And link it here.

**Comment by sindresorhus** [comment]
@UltCombo I went with documenting the caveat for now. Maybe we could use the node-glob `nodir` to only match files? Users could still match directories explicitly by using a trailing `/`.

**Comment by UltCombo** [comment]
> Maybe we could use the node-glob nodir to only match files? Users could still match directories explicitly by using a trailing `/`.

Not a bad idea at first glance, but that means use cases such as `del('node_modules')` will stop working, right?

**Comment by UltCombo** [comment]
Testing on Zsh, `dir1/**` does not match `dir1`.
screen shot 2015-07-27 at 6 01 22 pm

I wonder if it is a bug in node-glob. I'll open an issue when I get home.

**Comment by UltCombo** [comment]
I've opened an issue to track lone `**` in the last path segment: https://github.com/isaacs/node-glob/issues/213

Though, the [test case](https://github.com/sindresorhus/del/pull/24/files#diff-1dd241c4cd3fd1dd89c570cee98b79ddR135) that has been merged seems to be expected behavior from a globbing perspective. The `**` portion in `tmpDir/**/*` simply means deep (recursive), so it is equivalent to `tmpDir/*, tmpDir/*/*, tmpDir/*/*/* ...`

**Pull request closed without merging**